### ISSUE TITLE
Bug: can't build assimp for double precision on mac os

### ISSUE THREAD

**Issue by wbyates777** [assimp/assimp#5768]
assimp-5.4.3 does not build when ASSIMP_DOUBLE_PRECISION is ON on Mac Os Sonoma 14.6.1.

```
-- The C compiler identification is AppleClang 15.0.0.15000309
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Shared libraries enabled
-- compiling zlib from sources
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - failed
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- tinyusdz disabled
INFO Hunter enabled or RT-extension not found. glTF import/export will be built without Open3DGC-compression.
-- Enabled importer formats: AMF 3DS AC ASE ASSBIN B3D BVH COLLADA DXF CSM HMP IRRMESH IQM IRR LWO LWS M3D MD2 MD3 MD5 MDC MDL NFF NDO OFF OBJ OGRE OPENGEX PLY MS3D COB BLEND IFC XGL FBX Q3D Q3BSP RAW SIB SMD STL TERRAGEN 3D X X3D GLTF 3MF MMD
-- Disabled importer formats: USD
-- Enabled exporter formats: OBJ OPENGEX PLY 3DS ASSBIN ASSXML M3D COLLADA FBX STL X X3D GLTF 3MF PBRT ASSJSON STEP
-- Disabled exporter formats:
-- Treating all warnings as errors (for assimp library only)
-- Configuring done (2.3s)
-- Generating done (0.1s)
-- Build files have been written to: /Users/bill/Downloads/assimp-5.4.3
bill@CentralServer assimp-5.4.3 % make -j4
[  0%] Building C object contrib/zlib/CMakeFiles/zlibstatic.dir/adler32.c.o
[  1%] Building C object contrib/zlib/CMakeFiles/zlibstatic.dir/compress.c.o
[  1%] Building C object contrib/zlib/CMakeFiles/zlibstatic.dir/crc32.c.o
[  1%] Building C object contrib/zlib/CMakeFiles/zlibstatic.dir/deflate.c.o
/Users/bill/Downloads/assimp-5.4.3/contrib/zlib/compress.c:22:13: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
int ZEXPORT compress2(dest, destLen, source, sourceLen, level)
            ^
......<many screens of warnings>.....

           ^
/Users/bill/Downloads/assimp-5.4.3/contrib/zlib/trees.c:1168:12: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
local void bi_windup(s)
           ^
20 warnings generated.
[  5%] Linking C static library libzlibstatic.a
[  5%] Built target zlibstatic
[  5%] Building CXX object code/CMakeFiles/assimp.dir/Common/Assimp.cpp.o
[  5%] Building CXX object code/CMakeFiles/assimp.dir/CApi/CInterfaceIOWrapper.cpp.o
[  5%] Building CXX object code/CMakeFiles/assimp.dir/Common/Compression.cpp.o
[  6%] Building CXX object code/CMakeFiles/assimp.dir/Common/BaseImporter.cpp.o
[  6%] Building CXX object code/CMakeFiles/assimp.dir/Common/BaseProcess.cpp.o
In file included from /Users/bill/Downloads/assimp-5.4.3/code/Common/Assimp.cpp:50:
In file included from /Users/bill/Downloads/assimp-5.4.3/include/assimp/scene.h:58:
In file included from /Users/bill/Downloads/assimp-5.4.3/include/assimp/material.h:1700:
/Users/bill/Downloads/assimp-5.4.3/include/assimp/material.inl:65:38: error: out-of-line definition of 'GetTexture' does not match any declaration in 'aiMaterial'
AI_FORCE_INLINE aiReturn aiMaterial::GetTexture( aiTextureType type,
                                     ^~~~~~~~~~
/Users/bill/Downloads/assimp-5.4.3/include/assimp/material.inl:73:72: error: cannot initialize a parameter of type 'ai_real *' (aka 'double *') with an lvalue of type 'float *'
    return ::aiGetMaterialTexture(this,type,index,path,mapping,uvindex,blend,op,mapmode);
                                                                       ^~~~~
/Users/bill/Downloads/assimp-5.4.3/include/assimp/material.h:1680:18: note: passing argument to parameter 'blend' here
        ai_real *blend = NULL,
                 ^
In file included from /Users/bill/Downloads/assimp-5.4.3/code/Common/Assimp.cpp:50:
In file included from /Users/bill/Downloads/assimp-5.4.3/include/assimp/scene.h:58:
In file included from /Users/bill/Downloads/assimp-5.4.3/include/assimp/material.h:1700:
/Users/bill/Downloads/assimp-5.4.3/include/assimp/material.inl:162:75: error: cannot initialize a parameter of type 'float *' with an rvalue of type 'ai_real *' (aka 'double *')
            if (AI_SUCCESS == ::aiGetMaterialFloat(this, pKey, type, idx, &value)) {
                                                                          ^~~~~~
/Users/bill/Downloads/assimp-5.4.3/include/assimp/material.h:1564:16: note: passing argument to parameter 'pOut' here
        float *pOut) {
               ^


```

The error persists whether we build our own ZLIB or not.
I have version of zlib installed via macports.

**Comment by kimkulling** [comment]
Hi, the first warning was caused by the zlib: it uses a construct of local defined functions without marking them as static, if I get it right. I have opened the PR https://github.com/assimp/assimp/pull/5782 to mitigate the issue.

The second issue is caused by an inconsistent declaration of a material parameter. I am currenty trying to fix that issue. This was not detected because we currently not building the double precision feature by default.
